src/main/java/tic/tac/toe/Game.java

```diff
@@ -37,22 +58,32 @@ public char getCurrentPlayerSymbol() {
   }
 
   public void takeATurn() {
-    System.out.println(ui.prompt(currentPlayer.getSymbol()));
     currentPlayer.makeAMove(board);
-    if (currentPlayer.didMove()) {
-      changePlayer();
-    }
-    System.out.println(ui.displayBoard());
   }
 
   public void run() {
     boolean keepPlaying = true;
     System.out.println(ui.getGreeting());
-    System.out.println(ui.displayBoard());
+    System.out.println(ui.displayBoard(board));
     while (keepPlaying) {
-      takeATurn();
+      if (!isGameOver()) {
+        System.out.println(ui.prompt(currentPlayer.getSymbol()));
+        takeATurn();
+      }
+
+      if (isWinner()){
+        winningPlayer = currentPlayer;
+        System.out.println("Winner is " + currentPlayer.getSymbol());
+      }
+
+      if (currentPlayer.didMove()) {
+        changePlayer();
+      }
+
       keepPlaying = !isGameOver();
+      System.out.println("\n" + ui.displayBoard(board));
+      
     }
-    // insert output here of winner / tie state
+    System.out.println(ui.displayOutcome(winningPlayer == null ? ' ' : winningPlayer.getSymbol()));
```

Comment:
`' '` representing a tie game here kind of seems like magic to me. Could you think of a better way to model this?